orcavault/models/dcl/sal_library.sql

```diff
@@ -0,0 +1,70 @@
+{{
+    config(
+        indexes=[
+            {'columns': ['base_library_id'], 'type': 'btree'},
+            {'columns': ['alias_library_id'], 'type': 'btree'},
+        ],
+        materialized='incremental',
+        incremental_strategy='merge',
+        unique_key='sal_library_hk',
+        on_schema_change='fail'
+    )
+}}
+
+with source as (
+
+    select
+        *
+    from
+        {{ ref('hub_library') }}
+    {% if is_incremental() %}
+    where
+        cast(load_datetime as timestamptz) > ( select coalesce(max(load_datetime), '1900-01-01') as ldts from {{ this }} )
+    {% endif %}
+
+),
+
+filtered as (
+
+    select
+        library_hk as alias_library_hk,
+        library_id as alias_library_id,
+        (regexp_match(library_id, '(?:L\d{7}|L(?:PRJ|CCR|MDX|TGX)\d{6})'))[1] as base_library_id,
+        record_source
+    from
+        source
+    where
+        library_id like '%topup%' or library_id like '%rerun%'
```

Comment:
Why not use a "not regex match", e.g. `regex_like()`?
This would better match to the regex extraction of above and also cover cases that may not be topups or reruns (we may have has some of those in older data).

Reply:
Correct, yes. That's the intention. Hence, simple `like` syntax to scope them down. At the moment, I wish to start by limiting only to the library topup/rerun cases which are the minimal impact - data-wisely and, MVP.

If the solution works well then we can increase the radius. E.g. Any other base ID to another base ID linking cases. We need the Master Data Management process for that. So TBC.

I hope 🙏 we won't need it for the library. 🤞 _Or, this is something already had happened and you haven't told me yet!_

I will keep it simple `like` for now, Flo.

It (generally SAL tables are) is the business vault (purple) grade table. So, we can easily rebuild it, if rule change or business change request.